src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs

```diff
@@ -0,0 +1,1144 @@
+                    {
+                        throw new HttpRequestException("Saw CR without LF while parsing headers");
+                    }
+
+                    break;
+                }
+
+                // Get header name
+                while (c != ':')
+                {
+                    _sb.Append(c);
+                    c = await ReadCharAsync(cancellationToken);
+                }
+
+                string headerName = _sb.ToString();
+
+                _sb.Clear();
+
+                // Get header value
+                c = await ReadCharAsync(cancellationToken);
+                while (c == ' ')
+                {
+                    c = await ReadCharAsync(cancellationToken);
+                }
+
+                while (c != '\r')
+                {
+                    _sb.Append(c);
+                    c = await ReadCharAsync(cancellationToken);
+                }
+
+                if (await ReadCharAsync(cancellationToken) != '\n')
+                {
+                    throw new HttpRequestException("Saw CR without LF while parsing headers");
+                }
+
+                string headerValue = _sb.ToString();
+
+                // TryAddWithoutValidation will fail if the header name has trailing whitespace.
+                // So, trim it here.
+                // TODO: Not clear to me from the RFC that this is really correct; RFC seems to indicate this should be an error.
+                // However, tests claim this is important for compat in practice.
+                headerName = headerName.TrimEnd();
+
+                // Add header to appropriate collection
+                // Don't ask me why this is the right API to call, but apparently it is
+                if (!response.Headers.TryAddWithoutValidation(headerName, headerValue))
+                {
+                    if (!responseContent.Headers.TryAddWithoutValidation(headerName, headerValue))
+                    {
+                        // Header name or value validation failed.
+                        throw new HttpRequestException($"invalid response header, {headerName}: {headerValue}");
+                    }
+                }
+
+                _sb.Clear();
+
+                c = await ReadCharAsync(cancellationToken);
+            }
+
+            // Instantiate responseStream
+            HttpContentReadStream responseStream;
+
+            if (request.Method == HttpMethod.Head ||
+                status == 204 ||
+                status == 304)
+            {
+                // There is implicitly no response body
+                // TODO: I don't understand why there's any content here at all --
+                // i.e. why not just set response.Content = null?
+                // This is legal for request bodies (e.g. GET).
+                // However, setting response.Content = null causes a bunch of tests to fail.
+                responseStream = new ContentLengthReadStream(this, 0);
```

Comment:
I wasn't aware of this.

We should implement Expect: 100-continue support in the Managed handler, so hopefully that helps.